### PR TITLE
Fix completion

### LINKED ISSUE #11: Completion is bugged
So the completion is bugged now. At least in chat mode.

### PR DESCRIPTION
**Pull request by GroobleDierne**
Fix #11

### CODE REVIEW

**Comment by GroobleDierne** [comment]
There is still a bug, if you try to complete with no text it will work but you  won't be able to stop to complete

**Comment by GroobleDierne** [comment]
Okay it's done

File: index.js [hunk 1]
```diff
@@ -144,7 +144,7 @@ module.exports = function (options = {}) {
               return reject(err)
             }
 
-            resolve((matches || []).map(k => k.slice(string.length - string.lastIndexOf(' ') - 1)))
+            resolve((matches || []))
```
Comment (wvffle) [inline index.js]: There is one pair of unnecessary parenthesis left

File: index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const { listen: inputListen } = require('./src/input-manager')
 const modeManager = require('./src/mode-manager')
 const { commands, bind } = require('./src/commands')
 const { inspect } = require('util')
+const { match } = require('assert')
```
Comment (wvffle) [inline index.js]: Unused variable

File: index.js [hunk 3]
```diff
@@ -143,8 +144,10 @@ module.exports = function (options = {}) {
             if (err) {
               return reject(err)
             }
+            if (matches == null || matches[0] == undefined) resolve([])
```
Comment (wvffle) [inline index.js]: No newline before if, also I consider single line if statements without `{}` as unreadable in most cases

**Review by wvffle — CHANGES_REQUESTED** [review]

**Comment by wvffle** [comment]
Later today (maybe tomorrow) I'll add tests to check if completion really works

File: index.js [hunk 4]
```diff
@@ -143,8 +143,9 @@ module.exports = function (options = {}) {
             if (err) {
               return reject(err)
             }
-
-            resolve((matches || []).map(k => k.slice(string.length - string.lastIndexOf(' ') - 1)))
+            if (matches == null || matches[0] == undefined) { resolve([]) }
+            if (typeof matches[0].match === 'string') { resolve(matches.map(obj => obj.match)) }
```
Comment (wvffle) [inline index.js]: Is this required? when does it return an object?

Comment (wvffle) [reply index.js]: Okay, I see why

**Comment by wvffle** [comment]
Hmm, the completion seemed to be partially broken due to PrismarineJS/flying-squid#457 as I was testing using flying-squid

**Review by wvffle — APPROVED** [review]

**Pull request merged**